llvm/lib/Transforms/Scalar/ScalarizeMaskedMemIntrin.cpp

```diff
@@ -188,8 +191,39 @@ static void scalarizeMaskedLoad(const DataLayout &DL, CallInst *CI,
     return;
   }
 
+  // Optimize the case where the "masked load" is a predicated load - that is,
+  // where the mask is the splat of a non-constant scalar boolean. In that case,
+  // use that splated value as the guard on a conditional vector load.
+  if (isSplatValue(Mask, /*Index=*/0)) {
+    Value *Predicate = Builder.CreateExtractElement(Mask, uint64_t(0ull),
```

Comment:
Could you add a debug assert so that I can verify that this behavior worked without inspecting asm/ir ?

Reply:
I don't think so - we need the rest of the function

Maybe I could collect a statistic?